gammapy/data/observations.py

```diff
@@ -166,6 +168,8 @@ def available_irfs(self):
     @property
     def events(self):
         """Event list of the observation as an `~gammapy.data.EventList`."""
+        if self.obs_filter.time_filter is None and self._gti is not None:
+            self.obs_filter.time_filter = self._gti.time_intervals
```

Comment:
This line implements the feature mentionned in #5349. To my knowleged, there is no precision error

Reply:
Depends at which precision the events time is known/saved.

Reply:
The accuracy of the times of GTIs and events should be identical, so there is no in-depth reason to not have proper GTIs. One could add an accuracy tolerance, that I will set to 1/20 of the lowest psr period (20ms/10=1ms? is it enough?)

Reply:
Setting `interval_overlap_length(i1, i2) > 1e-3` in `check_time_intervals` seems to not fail right away but the loop over combinations of gti ` combinations(ti, 2)` is extremely long to compute because there is about 4e4 gtis in one obs so 16e8 combinations to test.

It should be part of the format convention that events are encompassed in the provided gti and done at the DL3 production level. I would argue that it is not the responsibility of gammapy to check that every time someone open a file, so by default we should not filter the events.

Reply:
I agree... even if I put it here for statistical reasons. It is up to us to decide.